### DISCUSSION TITLE
Citing R-Packages via "grateful::cite_packages()" breaks in pdf-output in quarto 1.4.549, but works in 1.3.450

### DISCUSSION THREAD

**Discussion by Gewerd-Strauss** [quarto-dev/quarto-cli#8580] [Q&A] [answered]
### Description

Hello.

I am quite a bit baffled. I recently updated to quarto v.1.4.549 - initially via `winget upgrade quarto -l .....`; and after investigating also via a manually downloaded installer. (Not that this should make any difference, but hey.). I initially noticed this in a project, and thus initially tested it in a dummy project as well. However, this is also reproducible for single documents.

# Setup

I have attached 4 sets of files, for the following scenarios:

- Quarto v.1.4.549 targeting a project, rendered to html/pdf/docx via `quarto render`. Format options are defined in `_quarto.yml`.
- Quarto v.1.4.549 targeting a single file, rendered to html/pdf/docx via  `quarto render about.qmd --to html,pdf,docx`
- Quarto v.1.3.450 targeting a project, rendered to html/pdf/docx via `quarto render`. Format options are defined in `_quarto.yml`.
- Quarto v.1.3.450 targeting a single file, rendered to html/pdf/docx via  `quarto render about.qmd --to html,pdf,docx`

Note that the file rendered is identical across all scenarios. While this means that the quarto v.1.3.450-cases contain breaking table captions, I considered equal input data more relevant in this scenario. Invalid table-captions should also not have anything to do with the issue itself.

No external citations are required. Packages are installed from renv for projects, and must be installed prior to use for standalone versions if not installed already.

---

# v.1.4.549

In both standalone and project setups, `grateful::cite_packages()` is successfully detecting packages loaded in the current session/all packages used within the project. 

In _HTML/DOCX_, `grateful::cite_packages()` resolves all packages successfully, with the exception of the package `renv`. I am not entirely sure why that is the case; but the bib-file generated by "grateful" does not contain a citation for it, so this makes at least some sense.

In PDF, all packages are added to the bibliography and displayed at the end of the file (again, with the exception of `renv`). However, their cite-keys within the generated tables are all mumbled strings. Interestingly, the _last_ entry of the table will get a citation. It is always wrong, but it resolves to this. Tests in other projects which I cannot share support this behaviour, regardless of which package name is listed last. And to make this more confusing, the version numbers for each package _are_ retrieved and attributed correctly.

# 1.3.450

In HTML/DOCX/PDF, all citations (except for `renv`) resolve successfully.

---

Example found in [this repo](https://github.com/Gewerd-Strauss/quarto-cli-discussion--8580)

---

I can't imagine what is causing this, but this is currently hard-breaking a project of mine; which is now retrospectively going to be downgraded to 1.3.450. Thankfully I had not yet transitioned much significant content to 1.4-features, so this at least saves me some time now.

I hope these details are sufficient to reproduce this problem. 

Thank you.  
Sincerely,

**Comment by mcanouil** [comment]
Please share Git repository to share code.
Thank you!

**Reply by Gewerd-Strauss** [reply]
Alright, I'll prep a git repo.

**Reply by Gewerd-Strauss** [reply]
The original post was updated.

**Reply by mcanouil** [reply]
Does the example have to be that big?
You are arguing the issue is also in simple document, can you share a simple Quarto document?

PS: you should be consistent with the syntax inside your documents, *i.e.*, not mixing, inline R code cell option with YAML style option and to not use `FALSE` (R) in YAML where it's `false`.
All this can confuse the parsers from Quarto/Pandoc and possibly the engine as well leading to unexpected results.
Also, `bibliography` is a global key, you don't have to duplicate it under each format.


Regarding, the fact that `renv` is not recorded, this has nothing to do with Quarto. Reach out to the authors of the R package and/or check the documentation.

Edit: I made a simple reproducible example and don't see any issue.

<table>
<tr><th>Quarto document</th><th>HTML</th></tr>
<tr>
<td>

````qmd
---
title: "Quarto Playground"
format: pdf
bibliography: "grateful-refs.bib"
---

This is a playground for Quarto.

```{r}
library(grateful)
library(knitr)
library(yaml)
```

```{r}
#| label: tbl-PackageVersions
knitr::kable(cite_packages(
  output = "table",
  out.dir = ".",
  cite.tidyverse = TRUE,
  pkgs = "All",
  include.RStudio = TRUE,
  dependencies = FALSE
))
```
````

</td>
<td>
<img width="1237" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/d2f298f8-12d5-4c1a-99ed-e247050fdea2">
</td>
</tr>
</table>

**Reply by Gewerd-Strauss** [reply]
Under which version are you running quarto? 

> You are arguing the issue is also in simple document, can you share a simple Quarto document?

I mean; I _did_ provide a single-file file which exhibits this issue in quarto v.1.4.549; and does _not_ do so in v.1.3.450. However, I am tring to simplify it now. 


> `renv`

I mentioned this merely to note that I found it odd that it didn't get cited by grateful not because I was thinking quarto to be at fault here, but didn't want to leave this outlier without comment because I am most definitely less capable of judging if there might be some interaction between `quarto` and `renv` than people working on the respective software.

---

I can successfully reproduce your input, which does not exhibit this issue on my system. With a bit more digging, I have also narrowed down what actually breaks on my end. See the below standalone file, converted via  `quarto render <name>.qmd --to pdf,html`


``````qmd
---
title: "reprex_new_standalone"
bibliography: "grateful-refs.bib"
---
1.4.549 project, rendered via `quarto render` (see `_quarto.yml`)
(table [-@tbl-PackageVersions])
(table [-@tbl-PackageVersions2])
(table [-@tbl-PackageVersions3])
This is a playground for Quarto.

```{r}
library(grateful)
library(knitr)
library(yaml)
cites <- cite_packages(
  output = "table",
  out.dir = ".",
  cite.tidyverse = TRUE,
  pkgs = "All",
  include.RStudio = TRUE,
  dependencies = FALSE
)
print(cites)
typeof(cites)
```

```{r}
#| label: tbl-PackageVersions
knitr::kable(cites)
```

```{r}
#| label: tbl-PackageVersions2
knitr::kable(cites,caption = "a caption for the table")
```

```{r}
#| label: tbl-PackageVersions3
#| tbl-cap: a caption for the table
knitr::kable(cites)
```

```{r}
sessionInfo()
```
``````

Note that I have changed the setup a bit; in that I first generate a generic var `cites` of type `list`, just so that I don't have to call `cite_packages()` multiple times. 

When converted to html, all three tables render successfully on my machine. When converting to pdf, only the first table gets rendered. The remaining get scrambled as shown in . The single difference between these tables is that the first does not contain a caption. The second and third one implement a caption either via the call to `knitr::kable()` or via the chunk-options. As documented and observable, this is required to be able to cross-reference a table. 

![grafik](https://github.com/quarto-dev/quarto-cli/assets/80548781/2c00956a-6f1f-4196-8341-b303e9dbbe93)


Note for clarity: The `grateful-refs.bib`-file is not included as it is not required, as it is generated by `cite_packages()`.


---

I hope this is reproducible now. 

---

P.S.: This is currently only tested to break on 1.4.549; mostly because I don't have access to the 1.3.450-installer currently and have merely enough bandwidth to write this text right now.

However I do not expect 1.3.450 to break where it so far works successfully.

**Reply by mcanouil** [reply]
> I mean; I did provide a single-file file which exhibits this issue in quarto v.1.4.549; and does not do so in v.1.3.450. However, I am tring to simplify it now.

To be fair, it's not a single file you provided (<https://github.com/Gewerd-Strauss/quarto-cli-discussion--8580/tree/master/quarto%201.4.549/grateful-package-issues>).
And not small either.

In the case of a cross-reference table (label prefix and caption), the following markdown and $LaTeX$ are produced (you can see that using `keep-md: true` and `keep-tex: true`):

<table>
<tr><th>Quarto document</th><th>HTML</th></tr>
<tr>
<td>

````md
::: {#tbl-PackageVersions .cell tbl-cap='a caption for the table'}

```{.r .cell-code}
knitr::kable(cites)
```

::: {.cell-output-display}


|Package   |Version |Citation                                       |
|:---------|:-------|:----------------------------------------------|
|base      |4.3.2   |@base                                          |
|devtools  |2.4.5   |@devtools                                      |
|knitr     |1.45    |@knitr2014; @knitr2015; @knitr2023             |
|prompt    |1.0.2   |@prompt                                        |
|renv      |1.0.3   |@renv                                          |
|rmarkdown |2.25    |@rmarkdown2018; @rmarkdown2020; @rmarkdown2023 |
|styler    |1.10.2  |@styler                                        |
|tidyverse |2.0.0   |@tidyverse                                     |
|yaml      |2.3.8   |@yaml                                          |


:::
:::
````

</td>
<td>

````latex
\begin{longtable}[]{@{}lll@{}}

\caption{\label{tbl-PackageVersions}a caption for the table}

\tabularnewline

\toprule\noalign{}
Package & Version & Citation \\
\midrule\noalign{}
\endhead
\bottomrule\noalign{}
\endlastfoot
base & 4.3.2 & 85b77c8a-261c-4f58-9b04-f21c67e0a758-1 \\
devtools & 2.4.5 & 85b77c8a-261c-4f58-9b04-f21c67e0a758-2 \\
knitr & 1.45 & 85b77c8a-261c-4f58-9b04-f21c67e0a758-3;
85b77c8a-261c-4f58-9b04-f21c67e0a758-4;
85b77c8a-261c-4f58-9b04-f21c67e0a758-5 \\
prompt & 1.0.2 & 85b77c8a-261c-4f58-9b04-f21c67e0a758-6 \\
renv & 1.0.3 & 85b77c8a-261c-4f58-9b04-f21c67e0a758-7 \\
rmarkdown & 2.25 & 85b77c8a-261c-4f58-9b04-f21c67e0a758-8;
85b77c8a-261c-4f58-9b04-f21c67e0a758-9;
85b77c8a-261c-4f58-9b04-f21c67e0a758-10 \\
styler & 1.10.2 & 85b77c8a-261c-4f58-9b04-f21c67e0a758-11 \\
tidyverse & 2.0.0 & 85b77c8a-261c-4f58-9b04-f21c67e0a758-12 \\
yaml & 2.3.8 & R Core Team (2023) \\

\end{longtable}
````

</td>
</tr>
</table>

Note that this only happens when it's a cross-reference (prefix `tbl` or whatever is a cross-reference in 1.4).
You can get the same behaviour with the following:

````qmd
---
title: "reprex_new_standalone"
bibliography: "grateful-refs.bib"
format: pdf
keep-md: true
keep-tex: true
---

```{r}
library(grateful)
library(knitr)
library(yaml)
cites <- cite_packages(
  output = "table",
  out.dir = ".",
  cite.tidyverse = TRUE,
  pkgs = "All",
  include.RStudio = TRUE,
  dependencies = FALSE
)
```

::: {#fig-PackageVersions}

```{r}
knitr::kable(cites)
```

A caption for a figure (but in fact it is a table).

:::
````

**Reply by mcanouil** [reply]
It appears, the issue is the interaction with fenced div surrounding the table.

@cscheid Any idea what is causing this in the new cross-reference system?

<table>
<tr><th>Bad</th><th>Good</th></tr>
<tr>
<td>

````qmd
---
format: pdf
---

::: {#tbl-PackageVersions tbl-cap='a caption for the table'}

::: {}

|Package   |Version |Citation   |
|:---------|:-------|:----------|
|base      |4.3.2   |@base      |
|devtools  |2.4.5   |@devtools  |
|prompt    |1.0.2   |@prompt    |
|renv      |1.0.3   |@renv      |
|styler    |1.10.2  |@styler    |
|tidyverse |2.0.0   |@tidyverse |
|yaml      |2.3.8   |@yaml      |

:::
:::

````

</td>
<td>

````qmd
---
format: pdf
---

::: {#tbl-PackageVersions tbl-cap='a caption for the table'}

|Package   |Version |Citation   |
|:---------|:-------|:----------|
|base      |4.3.2   |@base      |
|devtools  |2.4.5   |@devtools  |
|prompt    |1.0.2   |@prompt    |
|renv      |1.0.3   |@renv      |
|styler    |1.10.2  |@styler    |
|tidyverse |2.0.0   |@tidyverse |
|yaml      |2.3.8   |@yaml      |

:::
`````

</td>
</tr>
<tr><td>
<img width="358" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/78fa3268-d49e-4697-8a30-35efe792b939">
</td><td>
<img width="198" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/2c3c99ec-8373-4b09-af89-55c058b476d0">
</td></tr>
</table>

**Reply by Gewerd-Strauss** [reply]
> Note that this only happens when it's a cross-reference (prefix `tbl`).
> You can get the same behaviour with the following:

> ::: {#fig-PackageVersions}

I have no idea how to read this statement - as in "`fig-packageversions` does also _not_ work", or as in "`fig-packageversions` does resolve successfully". The latter would just be a workaround which - for obvious reasons - would still not be useful,

**Comment by Gewerd-Strauss** [comment]
> To be fair, it's not a single file you provided (https://github.com/Gewerd-Strauss/quarto-cli-discussion--8580/tree/master/quarto%201.4.549/grateful-package-issues).
> And not small either.


I _did_, [here (1.4.549)](https://github.com/Gewerd-Strauss/quarto-cli-discussion--8580/tree/master/quarto%201.4.549/grateful-package-issues-standalone) and [here (1.3.450)](https://github.com/Gewerd-Strauss/quarto-cli-discussion--8580/tree/master/quarto%201.3.450/grateful-package-issues-standalone), as I outlined in the issue itself:

> I initially noticed this in a project, and thus initially tested it in a dummy project as well. However, this is also reproducible for single documents.

&& 

> I have attached 4 sets of files, for the following scenarios:
> 
>     * Quarto v.1.4.549 targeting a project, rendered to html/pdf/docx via `quarto render`. Format options are defined in `_quarto.yml`.
> 
>     * Quarto v.1.4.549 targeting a single file, rendered to html/pdf/docx via  `quarto render about.qmd --to html,pdf,docx`
> 
>     * Quarto v.1.3.450 targeting a project, rendered to html/pdf/docx via `quarto render`. Format options are defined in `_quarto.yml`.
> 
>     * Quarto v.1.3.450 targeting a single file, rendered to html/pdf/docx via  `quarto render about.qmd --to html,pdf,docx`





Okay, to be fair I added 4 redundant `.bib`-files because I forgot that `grateful` generates that file itself.
The remaining files are output files I include to make sure any potential discrepancies that could arise between machines can be checked for even if the input file and processor seems to be identical.

The project-examples were included to be closer to the scenario which initially resulted in this issue on my side. I suppose as we can discard the project-scenario for now all project-related files become redundant as well.

**Reply by mcanouil** [reply]
Indeed, there were. I missed that one word difference in the sentences/commands.

May I suggest for the future to go straight to the point.
I mean by that here, you identified that the issue was in a standalone document for a single format. Simply share one single example using that format and mention that briefly "all formats worked with 1.3, all but PDF works with 1.4. Same behaviour observed with book project."

What usually works best when reporting issues and/or asking for help is a brief description and small self-contained reproducible example.

**Reply by Gewerd-Strauss** [reply]
Yea fair enough, that would have been the smarter way to go about it. I will keep it in mind for the future.

**Comment by cscheid** [comment]
> @cscheid Any idea what is causing this in the new cross-reference system?

That's a bug.

**Reply by mcanouil** [reply, marked as answer]
@Gewerd-Strauss The issue is now fixed and should be available in the future v1.5.9 pre-release build (not yet available).

- https://github.com/quarto-dev/quarto-cli/issues/8586

<img width="1624" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/79c1261a-8957-4b7a-8947-62f626f1340b">

**Reply by Gewerd-Strauss** [reply]
Alright, thank you for the ping. Will keep an eye out for the notification.

**Reply by Gewerd-Strauss** [reply]
Out of interest; how stable are pre-releases generally? 

I am asking because a stable `1.5` is probably not going to release "soon"; and thus I need to judge whether or not I will want to stay on stable `1.3.450` where I have a minimum set of required features supported; or if I want to write my projects with `1.4.X`-styling in mind by hopping to the (admittedly early)  pre-release of `1.5.9` (given that the stable 1.4-release `1.4.549` is incompatible in this one crucial-to-me scenario).
`1.4` offers quite a few things I would gladly incorporate and could use; but that would depend on how stable the `1.5`-prerelease is.

**Reply by mcanouil** [reply]
Well, it's a tricky question.
Pre-release means new things arrive so some breakage can also join the party.
Releases means it has been there without changes a bit longer than a pre-release.

I personally use the pre-release all the time in my work (to get all bug fixes when they arrive and get new features), but I am following closely updates so I know when to hold on a version and when I can update.

For example, my websites are rendered on CI regularly with a preview on PRs. I can see if there is some breaking or not before they arrive.
See the following issue that is on the current stable release:

- https://github.com/quarto-dev/quarto-cli/issues/8544

Now fixed but only on pre-release for now.


Note that I am less concerned about having the exact same HTML than having the exact same content.

**Reply by Gewerd-Strauss** [reply]
Alright, thank you for the insight. I think I will just have to try it out; see if I can get it working issue-free and then just stay at `1.5.9` before I transition my content to use `1.4` syntax; and if that is not the case I'll just have to stay on `1.3.450`.

**Comment by Gewerd-Strauss** [comment]
I installed `1.5.9` today, but this release seems to have introduced a couple bugs related to this. I will be splitting this into multiple sections; for clarity. I am not sure if this should become its own discussion, and am hesitant to open an additional issue myself. As the context of this discussion seems relevant I decided to post this follow-up here.

## General info
There are 2 main methods to caption a table: 

- `kable(object,caption = "caption")`
- via the cell-option `tbl-cap`

The presence of either one of these introduces the bugs described below; their erroneous intermediary outputs are seemingly identical. The example below is for captioning via `#| tbl-cap`, but note that captioning via `knitr::kable(...,caption="caption")` reproduces this issue 1:1 on my end.

When rendering to `pdf`,`docx` and `html`, only `pdf` contained the issues below.

Removing the `grateful-refs.bib` from the `bibliography`-YAML-key renders all citations as cite-keys à la `@citekey`. This can be used to confirm that the object generated by `grateful::cite_packages()` does assign keys correctly[^2]. This suggests that the issue lies somewhere beyond the scope of that package. However, while this intermediary is correct, this does not seem to be the version which ultimately gets rendered to pdf, as _that_ version has corrupted cite-keys in its table again[^3].


The `.bib`-file generated by `grateful:cite_packages()` (which is required for pandoc to resolve the citekeys) also lists all citekeys with their correct, unique, properly assigned citation metadata. Similarly, the bibliography inserted at file-end is built correctly, except for if we remove the  `bibliography`-YAML-key (in this case the bibliography is completely missing, which is expected behaviour).

---
---


Take the following document:

``````qmd
---
title: "reprex_new_standalone_overwriting_cites"
bibliography: "grateful-refs.bib"
format:
    pdf:
        keep-md: true
        keep-tex: true
    html:
        keep-md: true
    docx:
        keep-md: true
---
(table [-@tbl-PackageVersions2])
This is a playground for Quarto `r quarto::quarto_version()`.
Standalone file, rendered in RStudio via `Render Current Document`.

---


```{r}
library(grateful)
library(knitr)
library(sessioninfo)
library(formatdown)
library(processx)
library(quarto)
library(renv)
library(rmarkdown)
library(tidyverse)
library(yaml)
cites <- cite_packages(
  output = "table"
  , out.dir = "."
  , cite.tidyverse = TRUE
  , pkgs = "All"
  , include.RStudio = TRUE
  , dependencies = FALSE
  #, omit = c("base","formatdown")
)
print(cites)
typeof(cites)
```

---

caption via Cell label `#| tbl-cap`
<!-- caption via `knitr(...,caption = "caption")` -->


```{r}
#| label: tbl-PackageVersions2
#| echo: FALSE
#| warning: FALSE
#| cache: FALSE
#| tbl-cap: "caption via tbl-cap"
kable(cites)
```

--- 

manual table copied from the `.md`-intermediate: the cites themselves _can_ get rendered correctly.

|Package     |Version |Citation                                       |
|:-----------|:-------|:----------------------------------------------|
|base        |4.3.2   |@base                                          |
|formatdown  |0.1.2   |@formatdown                                    |
|knitr       |1.45    |@knitr2014; @knitr2015; @knitr2023             |
|processx    |3.8.3   |@processx                                      |
|quarto      |1.3     |@quarto                                        |
|renv        |1.0.3   |@renv                                          |
|rmarkdown   |2.25    |@rmarkdown2018; @rmarkdown2020; @rmarkdown2023 |
|sessioninfo |1.2.2   |@sessioninfo                                   |
|tidyverse   |2.0.0   |@tidyverse                                     |
|yaml        |2.3.8   |@yaml                                          |


---

```{r}
sessionInfo()
```
``````

This document consists of 2 tables, however the second one is merely the raw-markdown table resulting from conversion retrieved from the `file.pdf.md`-intermediary. This "output-table" remains the same if I convert to pdf, html or docx. I have checked that it does not change between output formats (at least, as far as the intermediary `.md`-file is concerned).

In PDF, the first citation in the table[^1] (in this scenario, `base`) will be inserted to various keys of citations thereafter, resulting in the following output:

1. Intermediary: 
![grafik](https://github.com/quarto-dev/quarto-cli/assets/80548781/b01a27ec-7653-4710-9b14-1e638a9fbccf)
2. PDF-Output:
![grafik](https://github.com/quarto-dev/quarto-cli/assets/80548781/2416c8a3-7e91-460e-aaf4-cfe4d550d8cc)

# Additional issues


While writing down this follow-up, I discovered a couple more things that go wrong.

1. Notice the disabled line `#, omit = c("base","formatdown")` in the call to `cite_packages()`? This argument lets the user omit packages which should not be cited. If we enable this line again (aka omit `base` and `formatdown`), the infringing package is no longer `base` or `formatdown`, but becomes `grateful`[^4]. This is because by default `grateful` omits itself; but overwriting the parameter `omit` without including `grateful` will reinsert it into the table; where it happens to take the first spot. You can go on with this, whatever package takes the first spot will be duplicated into all citations of the table after and including `rmarkdown2023`.
2. We can see that this issue always manifests in `@rmarkdown2023` and onwards. If we omit `rmarkdown`, this manifests starting at `sessionInfo`. If we also omit `sessionInfo`, this starts at `tidyverse`. And so on.
3. The two issues listed before can be combined. If we `omit` the top-most package in the table, the next-in-line will be duplicated while still exhibiting the behaviour described in `2.`.
4. I have no idea why it "breaks" on the citation it seems to break on.

---

The document used in this comment, alongside all rendered output-data & intermediaries can be found in the `1.5.9`-subfolder at https://github.com/Gewerd-Strauss/quarto-cli-discussion--8580.

Note that the changes I perform for additional details (e.g. removing yaml-keys) is not reflected, as this is only a single source-doc.

---


[^1]: See "Additional issues 1)" for details on why I specify "first citation"
[^2]: excerpt from pdf-md-intermediary `reprex_new_standalone.pdf.md`, as retrieved by setting `format.pdf.keep-md: true`:
![grafik](https://github.com/quarto-dev/quarto-cli/assets/80548781/20fb999f-9c4a-4904-8735-63b6bd306c5c)
[^3]: excerpt from the final pdf as retrieved if the YAML-key `bibliography` does not contain `grateful-refs.bib`. Note that the images in this footnote and footnote 1 are of the same execution.
 ![grafik](https://github.com/quarto-dev/quarto-cli/assets/80548781/6ab7585b-b3ae-4be8-8d50-c324928d10e6)
[^4]: excerpt from the final pdf as retrieved if grateful omits _only_ `base` and `formatdown`. `grateful` is no longer omitted, becomes the "first" package and is thus duplicated: 
![grafik](https://github.com/quarto-dev/quarto-cli/assets/80548781/9575fb40-f566-499a-afc8-70c56ae04218)
excerpt from pdf-md-intermediary:
![grafik](https://github.com/quarto-dev/quarto-cli/assets/80548781/3e61ba4c-34cb-4d76-8159-a4dbfc7a5aa9)

**Reply by cscheid** [reply]
I apologize for the continued problems on this part of the code. This is another bug. I'll fix right away. (For the morbidly curious, the bug happens when there are 10 or more wrapped citations, because the escaping produces prefixes of each other that then need to be processed in a specific order.)

**Reply by Gewerd-Strauss** [reply]
That's not a problem. I want to note that I do appreciate the fast response-time on this; particularly because I cannot imagine there to be ""nothing"" to do in projects of this scale - so... Thank you.

I mean... I know how long I need for resolving some of my own bugfixes, and that's nowhere near on the level of complexity as `quarto`.  Would feel kinda hypocritical to get annoyed here now.

---

Also: 

This second "follow-up" issue I posted - was it sufficient on details? Too overly detailed? Too broad? I am not the best at writing things _concisely_, so if I open issues they tend to be... big. 

As I have unfortunately also fallen back onto when raising issues/discussions here.

**Reply by mcanouil** [reply]
> This second "follow-up" issue I posted - was it sufficient on details? Too overly detailed? Too broad? I am not the best at writing things concisely, so if I open issues they tend to be... big.

Do not worry too much. We really appreciate the reports anyway. 😉
Carlos seems to have found what he needed.

**Reply by cscheid** [reply]
Your bug reports are fine, we much appreciate them!

The more concise you manage to get, the faster we are be able to pinpoint the problem. With that said, we prefer (by a very large margin!) long, good, complete, reproducible reports like yours over short, incomplete, and irreproducible ones.

**Reply by Gewerd-Strauss** [reply]
I can confirm that on 1.5.12 I can no longer reproduce this issue - neither in dedicated reprexes, nor in full-scale projects.

Thank you.
